### DISCUSSION TITLE
Should update_qdata start out false?

### DISCUSSION THREAD

**Discussion by tepperly** [LLNL/serac#817] [General]
I am looking at a LiDO run under valgrind, and valgrind is indicating that update_qdata is uninitialized. Should Functional's constructor initialize update_qdata to false? It only appears to be referenced in a handful of places. Based on what I see in the solid_functional, it looks like the default value should be false, but I am just guessing.

https://github.com/LLNL/serac/blob/6550835df4b05c97b79795be5d4507bcb9e24d3e/src/serac/numerics/functional/functional.hpp#L516

**Comment by samuelpmishLLNL** [comment]
Yes, it should be initialized in the constructor. It's an oversight on my part, I'll add a fix in one of our upcoming PRs. 

Fortunately, the only place `serac` currently uses `update_qdata` is in the solid mechanics module, where its value always set before use. So, it shouldn't be causing any problems yet, but it's better to remove the possibility!

**Reply by tepperly** [reply]
Great! It appears that one of our LiDO drivers is calling Mult before it's initialized leading to a valgrind reported conditional based on uninitialized value. What's the better initial value true or false?

**Reply by samuelpmishLLNL** [reply]
`False` by default makes the most sense to me.

**Reply by tepperly** [reply]
Thanks. By the way, we're experiencing some NaN's in one of our Serac-based LiDO tests. This could be our bad, but I've taken the step in our LiDO tests to turn quiet NaN's into signalling NaN's. If this is something you're interested in doing, it's done with
```c++
#include <cfenv>

main(int, char**)
{
      feenableexcept(FE_INVALID | FE_DIVBYZERO);
```

**Comment by tepperly** [comment]
```c++
diff --git a/src/serac/numerics/functional/functional.hpp b/src/serac/numerics/functional/functional.hpp
index d4a8222b..b6ec900f 100644
--- a/src/serac/numerics/functional/functional.hpp
+++ b/src/serac/numerics/functional/functional.hpp
@@ -202,7 +202,7 @@ public:
    */
   Functional(mfem::ParFiniteElementSpace*                               test_fes,
              std::array<mfem::ParFiniteElementSpace*, num_trial_spaces> trial_fes)
-      : test_space_(test_fes), trial_space_(trial_fes)
+      : update_qdata(false), test_space_(test_fes), trial_space_(trial_fes)
   {
     for (uint32_t i = 0; i < num_trial_spaces; i++) {
       P_trial_[i] = trial_space_[i]->GetProlongationMatrix();
```
